core/templates/dev/head/domain/exploration/ParamSpecObjectFactory.ts

```diff
@@ -66,9 +66,14 @@ export class ParamSpecObjectFactory {
   // keys which give tslint errors against underscore_casing in favor of
   // camelCasing.
   createFromBackendDict(paramSpecBackendDict: any): ParamSpec {
-    return new ParamSpec(
-      this.paramTypeObjectFactory.getTypeFromBackendName(
-        paramSpecBackendDict.obj_type));
+    // ts-ignore comment is required here since the function
+    // getTypeFromBackendName will throw and error if backendName
+    // is not registered for the object. So, this line results in a
+    // compilation error since ParamSpec constructor accepts only
+    // ParamType as argument but the passed argument has type ParamType | Error.
+    // @ts-ignore
```

Comment:
This seems odd to me. From the comment it sounds like the function *returns* an error, not throws one. Throwing should just cause an exception in the call stack and shouldn't need to be accounted for in the arg types. Am I misunderstanding something here?

Reply:
Yes, I think you are right. I missed that the function was just throwing an error and not returning one (the return type was specified incorrectly: `ParamType | Error`). This is fixed now.